test/conformance/api/v1/util.go

```diff
@@ -121,28 +121,27 @@ func validateDataPlane(t pkgTest.TLegacy, clients *test.Clients, names test.Reso
 // Validates the state of Configuration, Revision, and Route objects for a runLatest Service.
 // The checks in this method should be able to be performed at any point in a
 // runLatest Service's lifecycle so long as the service is in a "Ready" state.
-func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string, imagesForMultipleContainers ...map[string]string) error {
+func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string) error {
 	t.Log("Checking to ensure Revision is in desired state with", "generation", expectedGeneration)
 	err := v1test.CheckRevisionState(clients.ServingClient, names.Revision, func(r *v1.Revision) (bool, error) {
 		if ready, err := v1test.IsRevisionReady(r); !ready {
 			return false, fmt.Errorf("revision %s did not become ready to serve traffic: %w", names.Revision, err)
 		}
-		// for multi containers there will be more than one container and names.Image is a string,
-		// so in order to validate imageDigest for each images validateControlPlane accepting optional field called imagesForMultipleContainers
-		if names.Image == "" {
-			for _, value := range imagesForMultipleContainers {
+
+		if len(names.Images) == 1 {
+			if validDigest, err := shared.ValidateImageDigest(t, names.Images[0], r.Status.ContainerStatuses[0].ImageDigest); !validDigest {
```

Comment:
Im probably missing something obvious, but is this branch actually different from the `else` branch in the case where names.Images == 1? seems like in the case where len(names.Images) == 1 the else compares names.Images[0] to r.Status.ContainerStatuses[0].ImageDigest, which is the same thing this does. If so could we drop the if and just use the for loop for all cases?

Reply:
Refactored Please have a look

Reply:
looks better, though I'm trying to figure out if we definitely need to do extra work to match up the images: I'm pretty sure order is maintained in the resolver (see [this unit test](https://github.com/knative/serving/blob/master/pkg/reconciler/revision/background_test.go#L94) which validates that), if we could rely on that this could be a little simpler

Reply:
.. looks like even before the background resolution stuff landed we were maintaining order: https://github.com/knative/serving/blob/0882a7d27b75ce42e4d8b1213a42e87ab5aaa09e/pkg/reconciler/revision/revision.go#L120

Reply:
Yes you are right updated code